apps/prairielearn/src/sync/fromDisk/authors.sql

```diff
@@ -0,0 +1,94 @@
+-- BLOCK insert_authors
+INSERT INTO
+  authors (author_name, email, orcid, origin_course)
+SELECT
+  d.name,
+  d.email,
+  d.orcid,
+  d.origin_course::bigint
+FROM
+  (
+    SELECT DISTINCT
+      author.name,
+      author.email,
+      author.orcid,
+      author.origin_course
+    FROM
+      jsonb_to_recordset($authors::jsonb) author (
+        name text,
+        email text,
+        orcid text,
+        origin_course text
+      )
+  ) d
+  LEFT JOIN authors a ON (
+    a.author_name IS NOT DISTINCT FROM d.name
+    AND a.email IS NOT DISTINCT FROM d.email
+    AND a.orcid IS NOT DISTINCT FROM d.orcid
+    AND a.origin_course IS NOT DISTINCT FROM d.origin_course::bigint
+  )
```

Comment:
This is what lets us avoid incrementing the primary key sequence when we don't actually have any rows to insert, correct?